Mathlib/NumberTheory/FLT/Three.lean

```diff
@@ -16,6 +19,26 @@ The goal of this file is to prove Fermat Last theorem in the case `n = 3`.
 
 ## TODO
 Prove case 2.
+
+## Implementation details
+We follow the proof in <https://webusers.imj-prg.fr/~marc.hindry/Cours-arith.pdf>, page 43. The
+strategy is the following:
+* Case 1 is completely elementary and is proved using congruences modulo `9`.
+* To prove case 2, we consider the generalized equation `a ^ 3 + b ^ 3 = u * c ^ 3`, where `a`, `b`,
+  and `c` are in the cyclotomic ring `ℤ[ζ₃]` (where `ζ₃` is a primitive cube root of unity) and `u`
+  is a unit of `ℤ[ζ₃]`. `FermatLastTheoremForThree_of_FermatLastTheoremThreeGen` (whose proof is
+  rather elementary on paper) says that to prove Fermat's last theorem for exponent `3`, it is
+  enough to prove that this equation has no  solutions such that `c ≠ 0`, `¬ λ ∣ a`, `¬ λ ∣ b`,
+  `λ ∣ c` and `IsCoprime a b`. We call such a tuple a `Solution'`. A `Solution` is the same as a
+  `Solution'` with the additional assumption that `λ ^ 2 ∣ a + b`. We then prove that, given
+  `S' : Solution'`, there is `S : Solution` such that the multiplicity of `λ = ζ₃ - 1` in `c` is
+  the same in `S'` and `S`. In particular it is enough to prove that no `Solution` exists.
+  The key point is a descent argument on the multiplicity of `λ` in `c`: starting with
+  `S : Solution` we can find `S₁ : Solution` with multiplicity strictly smaller and this finishes
+  the proof. To construct `S₁` we go through a `Solution'` and then back to a `Solution`. More
+  importantly, we cannot control the unit `u`, and this is the reason why we need to consider
+  the generalized equation `a ^ 3 + b ^ 3 = u * c ^ 3`.
+
```

Comment:
Perhaps you could add a couple (not a thousand...) of pointers to the code, explaining how to bridge the informal-formal gap between your explanation above and what happens below.

Reply:
I've added a pointer to `exists_Solution_of_Solution'`. The rest of the argument will be formalized in the final PR and I will add more pointers.

Reply:
Thanks! Then I have nothing more to comment, I believe.